src/org/thoughtcrime/securesms/connect/AccountManager.java

```diff
@@ -29,80 +31,13 @@
-    private final static String DEFAULT_DB_NAME = "messenger.db";
-
-    public class Account {
-        private String dbName;
-        private String displayname;
-        private String addr;
-        private boolean configured;
-        private boolean current;
-
-        public String getDescr() {
-            String ret = "";
-            if (!displayname.isEmpty() && !addr.isEmpty()) {
-                ret = String.format("%s (%s)", displayname, addr);
-            } else if (!addr.isEmpty()) {
-                ret = addr;
-            } else {
-                ret = dbName;
-            }
-            if (!configured) {
-                ret += " (not configured)";
-            }
-            return ret;
-        }
-
-        public boolean isCurrent() {
-            return current;
-        }
-
-        public String getDbName() {
-            return dbName;
-        }
-    };
-
-    private @Nullable Account maybeGetAccount(File file) {
-        try {
-            if (!file.isDirectory() && file.getName().startsWith("messenger") && file.getName().endsWith(".db")) {
-                DcContext testContext = new DcContext(null, file.getAbsolutePath());
-                if (testContext.isOk()) {
-                    Account ret = new Account();
-                    ret.dbName = file.getName();
-                    ret.displayname = testContext.getConfig("displayname");
-                    ret.addr = testContext.getConfig("addr");
-                    ret.configured = testContext.isConfigured() != 0;
-                    return ret;
-                }
-            }
-        } catch (Exception e) {
-            e.printStackTrace();
-        }
-        return null;
-    }
-
-    private File getUniqueDbName(Context context) {
-        File dir = context.getFilesDir();
-        int index = 1;
-        while (true) {
-            File test = new File(dir, String.format("messenger-%d.db", index));
-            File testBlobdir = new File(dir, String.format("messenger-%d.db-blobs", index));
-            if (!test.exists() && !testBlobdir.exists()) {
-                return test;
-            }
-            index++;
-        }
-    }
-
     private void resetDcContext(Context context) {
-        // create an empty DcContext object - this will be set up then, starting with
-        // getSelectedAccount()
         ApplicationContext appContext = (ApplicationContext)context.getApplicationContext();
-        appContext.dcContext.notificationCenter.removeAllNotifiations();
-        appContext.dcContext.stopIo();
-        appContext.dcContext.unref();
-        appContext.dcContext = new ApplicationDcContext(context);
-        appContext.dcContext.setStockTranslations();
+        DcHelper.getNotificationCenter(context).removeAllNotifiations();
+        appContext.dcContext = appContext.dcAccounts.getSelectedAccount();
+        appContext.notificationCenter = new NotificationCenter(context);
+        appContext.eventCenter = new DcEventCenter(context);
```

Comment:
What does this line do? Remove all observers? Why do we need this?

Reply:
removing all observers is fine as also all activities are discarded. maybe it is not needed and the EventCenter and other objects can be persisted and longer-living, however, changing that is not in scope of this pr, i did not want to change these things and have a pretty narrow scope (the objets were recreated also previously)